accounts/abi/abi.go

```diff
@@ -87,7 +87,7 @@ func (abi ABI) getArguments(name string, data []byte) (Arguments, error) {
 	var args Arguments
 	if method, ok := abi.Methods[name]; ok {
 		if len(data)%32 != 0 {
-			return nil, fmt.Errorf("abi: improperly formatted output: %s - Bytes: [%+v]", string(data), data)
+			return nil, fmt.Errorf("abi: improperly formatted output: %x", data)
```

Comment:
Testded on the go playground: https://go.dev/play/p/raNSSxYMtHk
Yields

```
 s:  0@P`p�������� 
 v: [0 32 48 64 80 96 112 128 144 160 176 192 208 224 240]  
 x: 002030405060708090a0b0c0d0e0f0  
 q: "\x00 0@P`p\x80\x90\xa0\xb0\xc0\xd0\xe0\xf0"
```
In this case, if we want to keep it as close to the previous format as possible, I'd suggest this: 

```suggestion
			return nil, fmt.Errorf("abi: improperly formatted output: %q - Bytes: %+v", data, data)
```
It will print the string in a safe manner, and display the bytes like previously, but without double-brackets

Reply:
maybe %x with %v?

%q will do some additional interpretation, not sure if it's necessary/expected

Reply:
Well, currently, it does `%q` -- so "trying" to read the text as if it were ascii/unicode:ish text. With `%q`, that remains, so anything ascii-like will be shown, but non-printable characters will be changed ( in the playground-example above, `0@P'p` comes through. 

TLDR: is that `%q` is the correct way to display an untrusted/unsafe "string".

Reply:
I fail to see in which cases the `string(data)` is usefull (printed as `%s` or %q`); except to reduce changes. The `data` in question is almost always ABI encoded bytes returned by a contract, and here the error appears when it's not the format expected (eg. the wrong method was called, a revert message was sent back, etc.)

Reply:
Well, check the input to https://etherscan.io/tx/0x736162603c363154abb81571ee39c095a1e0a754fb26f83cf28c6207ffd8b1cb . There's even a telegram group (https://t.me/ethmsg) to read the plaintext-messages that are streamed over mainnet.

Reply:
Ehe, wasn't aware of the Telegram group :sweat_smile: 

However here it's the tx data. The error message in this PR is when a user does `abi.Unpack(name, data)` on some unexpected data. So I don't think printing `data` as a string is ever useful ; just saying.